lldb/packages/Python/lldbsuite/test/lang/swift/step_into_override/TestStepIntoOverride.py

```diff
@@ -1,16 +1,74 @@
-# TestStepIntoOverride.py
-#
-# This source file is part of the Swift.org open source project
-#
-# Copyright (c) 2014 - 2016 Apple Inc. and the Swift project authors
-# Licensed under Apache License v2.0 with Runtime Library Exception
-#
-# See https://swift.org/LICENSE.txt for license information
-# See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
-#
-# ------------------------------------------------------------------------------
-
-import lldbsuite.test.lldbinline as lldbinline
+import lldb
 from lldbsuite.test.decorators import *
+import lldbsuite.test.lldbtest as lldbtest
+import lldbsuite.test.lldbutil as lldbutil
+import os
+import platform
+import unittest2
 
-lldbinline.MakeInlineTest(__file__, globals(), decorators=[swiftTest])
+class TestStepIntoOverride(lldbtest.TestBase):
+    mydir = lldbtest.TestBase.compute_mydir(__file__)
+
+    @swiftTest
+    @skipIfLinux
+    def test_swift_stepping(self):
+        """Tests that we can step reliably in swift code."""
+        self.build()
+        self.do_test()
+
+    def setUp(self):
+        lldbtest.TestBase.setUp(self)
+        self.main_source = "main.swift"
+        self.main_source_spec = lldb.SBFileSpec(self.main_source)
+        # If you are running against a debug swift you are going to
+        # end up stepping into the stdlib and that will make stepping
+        # tests impossible to write.  So avoid that.
+
+        if platform.system() == 'Darwin':
+            lib_name = "libswiftCore.dylib"
+        else:
+            lib_name = "libswiftCore.so"
+
+        self.dbg.HandleCommand(
+            "settings set "
+            "target.process.thread.step-avoid-libraries {}".format(lib_name))
```

Comment:
I wonder if this is something we should do for all swift tests (or maybe just for all tests).  I can't think of a reason why we would want a test to have different behavior depending on whether you have line number information for the swift standard library.

Reply:
Mind if we do this as follow-up?

Reply:
That's fine.